contracts/ENSGovernorV2.sol

```diff
@@ -0,0 +1,187 @@
+
+        // Store the bond details
+        _proposalBonds[proposalId] = ProposalBond({
+            amount: bondAmount,
+            refunded: false,
+            proposer: msg.sender
+        });
+
+        emit BondCreated(msg.sender, bondAmount);
+    }
+
+    /**
+     * @dev Calculate the bond amount based on the count of targets.
+     * @param countOfTargets The number of targets in the proposal.
+     * @return The calculated bond amount.
+     */
+    function calculateBond(
+        uint256 countOfTargets
+    ) public view returns (uint256) {
+        return countOfTargets * ensPerTarget;
+    }
+
+    /**
+     * @dev Cancel a proposal and refund the bond if necessary.
+     * @param targets The addresses to call.
+     * @param values The values to send.
+     * @param calldatas The calldata to send.
+     * @param descriptionHash The hash of the proposal description.
+     * @return proposalId The ID of the proposal.
+     */
+    function _cancel(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) internal override(ENSGovernor) returns (uint256) {
+        uint256 proposalId = super._cancel(
+            targets,
+            values,
+            calldatas,
+            descriptionHash
+        );
+        (
+            uint256 againstVotesWithoutBond,
+            uint256 againstVotes, // votesLength
+            ,
+
+        ) = proposalVotes(proposalId);
+
+        ProposalBond storage bond = _proposalBonds[proposalId];
+
+        if (againstVotes >= againstVotesWithoutBond) {
+            require(
+                token.transfer(bond.proposer, bond.amount),
+                "ERC20: transferFrom failed"
+            );
+        } else {
+            availableBondsBalance += bond.amount;
+        }
+        lockedBondsBalance -= bond.amount;
+        return proposalId;
+    }
+
+    /**
+     * @dev Internal function to execute a proposal.
+     * Overrides the _execute function in ENSGovernor.
+     * Transfers the bond amount back to the proposer upon successful execution.
+     * @param proposalId The ID of the proposal to execute.
+     * @param targets The list of addresses the proposal calls.
+     * @param values The list of values to send with the calls.
+     * @param calldatas The list of call data to send with the calls.
+     * @param descriptionHash The hash of the proposal description.
+     */
+    function _execute(
+        uint256 proposalId,
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) internal override(ENSGovernor) {
+        ProposalBond storage bond = _proposalBonds[proposalId];
+
+        require(
+            token.transfer(bond.proposer, bond.amount),
+            "ERC20: transferFrom failed"
+        );
+        lockedBondsBalance -= bond.amount;
```

Comment:
The thing is that we establish price (uint256 public bondPricePerTarget = 1 ether;) in contract so that it's not possible do not have a bond. Or I may change this logic somehow?